src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -596,42 +586,49 @@ class ReferenceProcessorAtomicMutator: StackObj {
   }
 };
 
-// This class is an interface used to implement task execution for the
-// reference processing.
-class AbstractRefProcTaskExecutor {
-public:
-
-  // Abstract tasks to execute.
-  class ProcessTask;
+enum class RefProcThreadModel { Multi, Single };
 
-  // Executes a task using worker threads.
-  virtual void execute(ProcessTask& task, uint ergo_workers) = 0;
-};
-
-// Abstract reference processing task to execute.
-class AbstractRefProcTaskExecutor::ProcessTask {
+class RefProcTask : StackObj {
```

Comment:
Maybe put some comments here, explaining what `RefProcTask` and `RefProcProxyTask` are, their interaction/relation.

Reply:
I will try to do that.